recipes/proj/all/conanfile.py

```diff
@@ -30,162 +38,137 @@ class ProjConan(ConanFile):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
 
     def _patch_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
+        apply_conandata_patches(self)
+
+        cmakelists = os.path.join(self.source_folder, "CMakeLists.txt")
+        replace_in_file(self, cmakelists, "/W4", "")
 
-        cmakelists = os.path.join(self._source_subfolder, "CMakeLists.txt")
-        tools.replace_in_file(cmakelists, "/W4", "")
+        # Fix up usage of SQLite3 finder outputs
+        rm(self, "FindSqlite3.cmake", os.path.join(self.source_folder, "cmake"))
+        replace_in_file(self, cmakelists, "SQLITE3_FOUND", "SQLite3_FOUND")
+        replace_in_file(self, cmakelists, "SQLITE3_VERSION", "SQLite3_VERSION")
+        replace_in_file(self, cmakelists, "find_package(Sqlite3 REQUIRED)", "find_package(SQLite3 REQUIRED MODULE)")
 
         # Let CMake install shared lib with a clean rpath !
-        if tools.Version(self.version) >= "7.1.0" and tools.Version(self.version) < "9.0.0":
-            tools.replace_in_file(cmakelists,
+        if Version(self.version) >= "7.1.0" and Version(self.version) < "9.0.0":
+            replace_in_file(self, cmakelists,
                                   "set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)",
                                   "")
 
-        # Trick to find sqlite3 executable for build machine
-        # TODO: shouldn't be necessary in conan v2 with VirtualBuildEnv?
-        sqlite3_exe = " ".join("\"{}\"".format(path.replace("\\", "/")) for path in self.deps_env_info["sqlite3"].PATH)
-        tools.replace_in_file(
-            cmakelists,
-            "find_program(EXE_SQLITE3 sqlite3)",
-            "find_program(EXE_SQLITE3 sqlite3 PATHS {} NO_DEFAULT_PATH)".format(sqlite3_exe),
-        )
-
-        # Agressive workaround against SIP on macOS, to handle sqlite3 executable
-        # linked to shared sqlite3 lib
-        if tools.is_apple_os(self._settings_build.os):
-            # TODO: no hope for 2 profiles, wait for stable self.dependencies
-            #       because we want absolute lib paths of build profile actually
-            if not hasattr(self, "settings_build"):
-                if tools.Version(self.version) < "8.1.0":
-                    cmake_sqlite_call = "CMakeLists.txt"
-                    pattern = "${EXE_SQLITE3}"
-                else:
-                    cmake_sqlite_call = "generate_proj_db.cmake"
-                    pattern = "\"${EXE_SQLITE3}\""
-                lib_paths = self.deps_cpp_info["sqlite3"].lib_paths
-                tools.replace_in_file(
-                    os.path.join(self._source_subfolder, "data", cmake_sqlite_call),
-                    "COMMAND {}".format(pattern),
-                    "COMMAND ${{CMAKE_COMMAND}} -E env \"DYLD_LIBRARY_PATH={}\" {}".format(
-                        ":".join(lib_paths), pattern
-                    ),
-                )
```

Comment:
No it's not what I meant. I said that the block under `if not hasattr(self, "settings_build")` can be applied unconditionally:

```python
        # Agressive workaround against SIP on macOS, to handle sqlite3 executable
        # linked to shared sqlite3 lib
        if is_apple_os(self):
            if Version(self.version) < "8.1.0":
                cmake_sqlite_call = "CMakeLists.txt"
                pattern = "${EXE_SQLITE3}"
            else:
                cmake_sqlite_call = "generate_proj_db.cmake"
                pattern = "\"${EXE_SQLITE3}\""
            lib_paths = self.dependencies["sqlite3"].cpp_info.libdirs
            replace_in_file(
                self,
                os.path.join(self.source_folder, "data", cmake_sqlite_call),
                "COMMAND {}".format(pattern),
                "COMMAND ${{CMAKE_COMMAND}} -E env \"DYLD_LIBRARY_PATH={}\" {}".format(
                    ":".join(lib_paths), pattern
                ),
            )
```

Reply:
ah whoops, thanks